kobo-uncaged/device/device.go

```diff
@@ -463,90 +422,116 @@ func (k *Kobo) readMDfile() error {
 		dbMimeType   string
 		dbFileSize   int
 	)
-	query := `
-		SELECT ContentID, Title, Attribution, Description, Publisher, Series, SeriesNumber, MimeType, ___FileSize 
-		FROM content
+	queryFrom := ` FROM content
 		WHERE ContentType=6
 		AND MimeType NOT LIKE 'image%%'
 		AND (IsDownloaded='true' OR IsDownloaded=1)
 		AND ___FileSize>0
 		AND Accessibility=-1
 		AND ContentID LIKE ?;`
```

Comment:
This query is rather slow (it seems to take several seconds on my test kobo with ~700 books). It may be a bit too specific and could be relaxed a bit.

Do you have any ideas @pgaskin? I'm basically trying to get sideloaded books only from the DB.

Reply:
You'll probably need to experiment with it to make any conclusions on performance. I'd recommend testing it on a computer.

For simplifying it, I wonder if it'd be faster to look only at the content ID for the path prefix, then to filter it in KU from there.

Reply:
I was just thinking that myself.

And it probably doesn't matter if I overestimate the number of books (and therefore over allocate the metadata map), because it may mean less memory is allocated when adding new books later.

Reply:
Interesting. Using `SELECT COUNT(ContentID) FROM content WHERE ContentType=6;` is just as slow as the previous query.

Let's see what happens without any conditions at all.

Reply:
Not worrying about this for now. Can't seem to improve it easily.